Packs/CyberArkEventCollector/Integrations/CyberArkEventCollector/CyberArkEventCollector.py

```diff
@@ -0,0 +1,153 @@
+# pylint: disable=no-name-in-module
+# pylint: disable=no-self-argument
+import urllib3
+from SiemApiModule import *  # noqa: E402
+
+urllib3.disable_warnings()
+
+# -----------------------------------------  GLOBAL VARIABLES  -----------------------------------------
+DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%f'
+EVENT_FIELDS = [
+    'ID',
+    'AuthMethod',
+    'DirectoryServiceUuid',
+    'DirectoryServicePartnerName',
+    'EntityName',
+    'EntityType',
+    'EntityUuid',
+    'FromIPAddress',
+    'Level',
+    'ImpersonatorUuid',
+    'NewEntity',
+    'NormalizedUser',
+    'OldEntity',
+    'RequestDeviceOS',
+    'RequestHostName',
+    'RequestIsMobileDevice',
+    'Tenant',
+    'UserGuid',
+    'WhenLogged',
+    'WhenOccurred',
+]
+
+
+class CyberArkEventsRequest(IntegrationHTTPRequest):
+    method = Method.GET
+    headers = {'Accept': '*/*', 'Content-Type': 'application/json'}
+
+
+class CyberArkEventsClient(IntegrationEventsClient):
+    request: IntegrationHTTPRequest
+    options: IntegrationOptions
+
+    def __init__(
+        self,
+        request: CyberArkEventsRequest,
+        options: IntegrationOptions,
+        credentials: Credentials,
+        session=requests.Session(),
+    ) -> None:
+        self.access_token = None
+        self.credentials = credentials
+        super().__init__(request, options, session)
+
+    def set_request_filter(self, after: Any):
+        return after
+
+    def authenticate(self):
+        credentials = base64.b64encode(f'{self.credentials.identifier}:{self.credentials.password}'.encode()).decode()
+        request = IntegrationHTTPRequest(
+            method=self.request.method,
+            url=f"{demisto.params().get('url', '').removesuffix('/')}/oauth2/token/{demisto.params().get('app_id')}",
```

Comment:
pass the fields to the constructor instead of accessing them using `demisto.params`